waspc/src/Wasp/Generator/DbGenerator/Operations.hs

```diff
@@ -20,13 +28,25 @@ printJobMsgsUntilExitReceived chan = do
     J.JobOutput {} -> printJobMessage jobMsg >> printJobMsgsUntilExitReceived chan
     J.JobExit {} -> return ()
 
-migrateDev :: Path' Abs (Dir ProjectRootDir) -> IO (Either String ())
-migrateDev projectDir = do
+-- | Migrates in the generated project context and then copies the migrations dir back
+-- up to the Wasp source dir to ensure they remain in sync.
+migrateDevAndSync :: Path' Abs (Dir WaspProjectDir) -> Path' Abs (Dir ProjectRootDir) -> IO (Either String ())
+migrateDevAndSync waspDir projectDir = do
   chan <- newChan
   (_, dbExitCode) <-
     concurrently
       (printJobMsgsUntilExitReceived chan)
       (DbJobs.migrateDev projectDir chan)
   case dbExitCode of
-    ExitSuccess -> return (Right ())
+    ExitSuccess -> copyMigrationsUp waspDir projectDir
     ExitFailure code -> return $ Left $ "Migrate (dev) failed with exit code: " ++ show code
+
+copyMigrationsUp :: Path' Abs (Dir WaspProjectDir) -> Path' Abs (Dir ProjectRootDir) -> IO (Either String ())
```

Comment:
```suggestion
copyMigrationsFromGenProjectDirToWaspProjectDir :: Path' Abs (Dir WaspProjectDir) -> Path' Abs (Dir ProjectRootDir) -> IO (Either String ())
```
I tried to make it more explicit but now we have a monster :D. I like the terminology of "up" and "down", sounds like something might be useful in other instances too. Maybe you could just add a comment above the function explaining what "up" means?

Also, maybe it would make sense to flip the order of the input args, so we have (src -> target) instead of (target -> src) ordering.

Reply:
Good call! 👍🏻 I can rif on the name some, or at least comment better, and have it be (src -> target) intead

Reply:
Decided to go with the following based on your notes, where I updated the name, added a comment, and swapped ordering:

```haskell
-- | Copies the DB migrations from the generated project dir back up to theh wasp project dir
copyMigrationsBackToSource :: Path' Abs (Dir ProjectRootDir) -> Path' Abs (Dir DbMigrationsDir) -> IO (Either String ())
```

Thanks!